### ISSUE TITLE
How to block NS from updating core modules and tns android

### ISSUE THREAD

**Issue by terhoraj** [NativeScript/NativeScript#3561]
Got nativescript 2.5 today which broke everything. I want to be able to block completely all these updates since they provide more bad things than good things. I have already used too much time in work trying to prepare all this made..

**Comment by terhoraj** [comment]
These updates are serious problem in production environment.

**Comment by terhoraj** [comment]
For anyone getting ruined product via this update go revert everything using these instructions:
http://fluentreports.com/blog/?p=303
and remove Android sdk 25, it's buggy.

**Comment by tsonevn** [comment]
Hi @terhoraj,
You could change versions for the `tns-core-modules` and `tns-android` in your project `package.json` file manually. You should also remove the Caret `^` and Tilda `~` symbols before the versions. Your `package.json` file should look like the example below.

`package.json`
```
{
  "nativescript": {
    "id": "org.nativescript.sidedrawerTapItem",
    "tns-android": {
      "version": "2.4.1"
    },
    "tns-ios": {
      "version": "2.4.0"
    }
  },
  "dependencies": {
    "nativescript-telerik-ui": "1.5.1",
    "tns-core-modules": "2.4.3"
  },
  "devDependencies": {
    ...
  }
}
```

You should install also the old `CLI`, which will prevent you from some errors from using old modules and new `CLI`.
To install old CLI, you could follow the below-given commands.

1. `npm uninstall nativescript -g`
2. `npm install nativescript@<version> -g`

Regarding the issue, could you provide some more info about the problem, you are having. This will help us to provide a better solution for your case.

**Comment by abinici** [comment]
After the upgrade of NS to version 2.5, I began experiencing sudden crashes of my app even though the app is relatively small at this point.

The app starts up and shows the initial screen, which only consists of a logo, search field and a button.   And there is no I/O activity at this point.

Despite the simplicity of the initial screen and absence of I/O + user interaction, the app crashes after a few seconds and restarts. I confirm that this happens with the iOS Simulator. I haven't tried with Android.

This is what is printed out in the shell just before the crash:

```
8:06:59 PM - Compilation complete. Watching for file changes.

Feb  2 20:07:03 BinMac com.apple.CoreSimulator.SimDevice.BA2C79D3-C503-4596-8604-8EBECA0D1594.launchd_sim[2010] (UIKitApplication:com.some.app[0xa0c6][2194]): Service exited due to Killed: 9
```

This did not happened with the previous version of NS (2.4). I will have to downgrade until this gets fixed.

Here is the output of `tns info`:

```
│ Component        │ Current version            │ Latest version │ Information │
│ nativescript     │ 2.6.0-2017-02-02-7897      │ 2.5.0          │ Up to date  │
│ tns-core-modules │ 2.5.0                      │ 2.5.0          │ Up to date  │
│ tns-android      │ 2.6.0-next-2017-02-01-1648 │ 2.5.0          │ Up to date  │
│ tns-ios          │ 2.6.0-2017-2-2-1           │ 2.5.0          │ Up to date 
```

**Comment by Daxito** [comment]
It is interesting that you say you are using NS 2.5 but your current version is 2.6.0-2017-02-02-7897 ...

**Comment by abinici** [comment]
@Daxito, you are right. When I downgraded NS to 2.5, the crashes stopped. Thank you for pointing out the obvious :)
